modules/libcom/src/osi/os/RTEMS-posix/osdEvent.c

```diff
@@ -0,0 +1,111 @@
+/*************************************************************************\
+* Copyright (c) 2021 Fritz Haber Institute, Berlin
+* SPDX-License-Identifier: EPICS
+* EPICS BASE is distributed subject to a Software License Agreement found
+* in file LICENSE that is included with this distribution.
+\*************************************************************************/
+/*
+ * RTEMS osdEvent.c
+ *      Author: H. Junkes
+ *              junkes@fhi.mpg.de
+ */
+
+#include <stdio.h>
+#include <malloc.h>
+#include <limits.h>
+#include <rtems.h>
+#include <rtems/error.h>
+
+#include <rtems/thread.h>
+
+#include "libComAPI.h"
+#include "epicsEvent.h"
+
+typedef struct epicsEventOSD {
+    rtems_binary_semaphore rbs;
+} epicsEventOSD;
+
+/*
+ * Create a simple binary semaphore
+ */
+LIBCOM_API epicsEventId
+epicsEventCreate(epicsEventInitialState initialState)
+{
+    epicsEventOSD *pSem = malloc (sizeof(*pSem));
+
+    if (pSem) {
+        rtems_binary_semaphore_init(&pSem->rbs, NULL);
+        if (initialState)
+            rtems_binary_semaphore_post(&pSem->rbs);
+    }
+    return pSem;
+}
+
+LIBCOM_API void
+epicsEventDestroy(epicsEventId pSem)
+{
+    rtems_binary_semaphore_destroy(&pSem->rbs);
+}
+
+LIBCOM_API epicsEventStatus
+epicsEventTrigger(epicsEventId pSem)
+{
+    rtems_binary_semaphore_post(&pSem->rbs);
+    return epicsEventOK;
+}
+
+LIBCOM_API epicsEventStatus
+epicsEventWait(epicsEventId pSem)
+{
+    rtems_binary_semaphore_wait(&pSem->rbs);
+    return epicsEventOK;
+}
+
+LIBCOM_API epicsEventStatus
+epicsEventWaitWithTimeout(epicsEventId pSem, double timeOut)
+{
+    int sc;
+    rtems_interval delay;
+    int rate = rtems_clock_get_ticks_per_second();
```

Comment:
As I read it, `rtems_clock_get_ticks_per_second()` will evaluate as `uint32_t`.  The `tick` argument of `rtems_binary_semaphore_wait_timed_ticks()` is also `uint32_t`.  I don't think the mixing of signed int here would cause a problem, but it certainly doesn't help.